### DISCUSSION TITLE
Razer Synapse instruction

### DISCUSSION THREAD

**Discussion by CPTVICKERS** [ixjf/MSIRGB#200] [Ideas]
I'm an idiot from the internet. Is there a way to make MSIRGB listen to Razer synapse for direction on what to do? MSIRGB has proven to be 1000% better than the garbage bloatware MSI has pushed out. It looks like MSI attempted to cut costs, by making only one program that worries about ARGB and abandon actual support for all cheaper RGB only boards. Mystic Light works about as well as swimming in tar, and dragon whatever doesn't sync my board to razer synapse. I would rather razer sync MSIRGB since it works.

I tried nosing around in synapse, but it appears it is written in C. I know nothing of C, and can barely lua. If this is impossible, no will suffice.

**Comment by ixjf** [comment]
You could probably make a Lua script to do this, indeed, but I can't do this myself as I don't have the hardware, and apparently you don't have the knowledge either, so maybe look into OpenRGB? That might do what you want.

**Comment by CPTVICKERS** [comment]
Thanks, it's possible gives me hope. Maybe I learn something new, or if
openRGB has anything, I'll use that.

**Comment by jsphcsnrs** [comment]
OpenRGB doesn't have great support for the RGB-only MSI boards though. Just tried it and it only allows static colors (at least for my Z390-A Pro).

**Reply by ixjf** [reply]
Hmm, was not aware of that.

**Comment by CPTVICKERS** [comment]
I have a B450 tomahawk, it's a great board, and I'm og (old gamer) so I
didn't buy the board because it had lights on it, I bought it to build
within budget, for specs n features as needed. The manufacturers just snuck
the RGB in on me. My Gskill ram dollar for performance, happened to have
ARGB. (bought on some sort of sale with some sort of instant rebate) I
currently have a MSI 2080 x gaming, guess what, more ARGB. I made it all
somewhat synced with synapse (the board can only control RGB on itself, or
externally for lack of ARGB controller) and for some reason, I started to
care about blinky colorful things.

This is also the first year I put my rig on a wall shelf, where it can
actually be seen. (woman and cats make reasons) Every time my PC sleeps or
restarts, all the ARGB/RGB in the case loses sync, and stays one color.
Furthermore, without force quitting MSI's garbage, and restarting, it will
not reinitialize control. Then after all that, synapse may or may not need
to be restarted. So when I found your simple program, I was thrilled,
because it just works. I even adjusted the police light script to reflect
my local PD's colors and frequency, which to my delight, the program read
the script every time I clicked on it. That I loved, because I didn't have
to reload the program, to reload scripts, which allowed for fast editing.

I don't have the knowledge, but I wrote my own scripts for m.u.d. games
back in the day, and edditied a few for addons along the way. I forgot most
of what I learned for lack of use in ohh, 10 years or so. I don't expect to
just jump right into it, and make it work, especially since I have never
interfaced with hardware before. This is something that is interesting and
possibly fun to learn about.

Thank you for your work, and taking the time to respond. You are awesome.
